routers/api/v1/repo/commits.go

```diff
@@ -149,46 +153,73 @@ func GetAllCommits(ctx *context.APIContext) {
 	}
 
 	sha := ctx.FormString("sha")
+	path := ctx.FormString("path")
+
+	var (
+		commitsCountTotal int64
+		commits           []*git.Commit
+	)
+
+	if len(path) == 0 {
+		var baseCommit *git.Commit
+		if len(sha) == 0 {
+			// no sha supplied - use default branch
+			head, err := gitRepo.GetHEADBranch()
+			if err != nil {
+				ctx.Error(http.StatusInternalServerError, "GetHEADBranch", err)
+				return
+			}
+
+			baseCommit, err = gitRepo.GetBranchCommit(head.Name)
+			if err != nil {
+				ctx.Error(http.StatusInternalServerError, "GetCommit", err)
+				return
+			}
+		} else {
+			// get commit specified by sha
+			baseCommit, err = gitRepo.GetCommit(sha)
+			if err != nil {
+				ctx.Error(http.StatusInternalServerError, "GetCommit", err)
+				return
+			}
+		}
 
-	var baseCommit *git.Commit
-	if len(sha) == 0 {
-		// no sha supplied - use default branch
-		head, err := gitRepo.GetHEADBranch()
+		// Total commit count
+		commitsCountTotal, err = baseCommit.CommitsCount()
 		if err != nil {
-			ctx.Error(http.StatusInternalServerError, "GetHEADBranch", err)
+			ctx.Error(http.StatusInternalServerError, "GetCommitsCount", err)
 			return
 		}
 
-		baseCommit, err = gitRepo.GetBranchCommit(head.Name)
+		// Query commits
+		commits, err = baseCommit.CommitsByRange(listOptions.Page, listOptions.PageSize)
 		if err != nil {
-			ctx.Error(http.StatusInternalServerError, "GetCommit", err)
+			ctx.Error(http.StatusInternalServerError, "CommitsByRange", err)
 			return
 		}
 	} else {
-		// get commit specified by sha
-		baseCommit, err = gitRepo.GetCommit(sha)
+		if len(sha) == 0 {
+			sha = ctx.Repo.Repository.DefaultBranch
```

Comment:
Still use `head, err := gitRepo.GetHEADBranch()`?

Reply:
I don't think so.
1. it always returns an empty list, if without `sha` or with does not change this (and it doesn't work with pure SHAs or with branch names).
2. tested it with and it does not work